src/com/seafile/seadroid2/ui/activity/BrowserActivity.java

```diff
@@ -948,6 +943,90 @@ public void onFileSelected(SeafDirent dirent) {
         startFileActivity(repoName, repoID, filePath);
     }
 
+    public void downloadDir(String direntName) {
+        if (!Utils.isNetworkOn()) {
+            showToast(R.string.network_down);
+            return;
+        }
+
+        final String repoName = navContext.getRepoName();
+        final String repoID = navContext.getRepoID();
+        final String filePath = Utils.pathJoin(navContext.getDirPath(), direntName);
+        ConcurrentAsyncTask.execute(new DownloadDirTask(), repoName, repoID, filePath);
+        // Log.d(DEBUG_TAG, "download >> " + repoName + navContext.getDirPath());
+
+    }
+
+    private class DownloadDirTask extends AsyncTask<String, Void, List<SeafDirent> > {
+
+        String repoName;
+        String repoID;
+        String filePath;
+
+        SeafException err = null;
+        
+        @Override
+        protected List<SeafDirent> doInBackground(String... params) {
+            if (params.length != 3) {
+                Log.d(DEBUG_TAG, "Wrong params to LoadDirTask");
+                return null;
+            }
+
+            repoName = params[0];
+            repoID = params[1];
+            filePath = params[2];
+
+            List<SeafDirent> dirents;
+            try {
+                dirents = dataManager.getDirentsFromServer(repoID, filePath);
+            } catch (SeafException e) {
+                err = e;
+                e.printStackTrace();
+                return null;
+            }
+            
+            if (dirents == null) 
+            	return null;
+
+            for (SeafDirent seafDirent : dirents) {
+                if (!seafDirent.isDir()) {
+                    File localCachedFile = dataManager.getLocalCachedFile(repoName, 
+                                                                          repoID, 
+                                                                          Utils.pathJoin(filePath, 
+                                                                                          seafDirent.name), 
+                                                                          seafDirent.id);
+                    if (localCachedFile != null) {
+                    	continue;
+                	} 
+
+                    // Log.d(DEBUG_TAG, Utils.pathJoin(repoName, filePath, seafDirent.name));
+                    txService.addTaskToDownloadQue(account,
+                            repoName,
+                            repoID,
+                            Utils.pathJoin(filePath,
+                                    seafDirent.name));
+                }
+
```

Comment:
After adding download tasks, we should show a toast "starting to download 10 files". Or if all files is cached, show something like "All files has been downloaded".